### ISSUE TITLE
Plugin slowly falls behind

### ISSUE THREAD

**Issue by jcornett5** [Loze-Put/hunllef-helper#31]
During the fight the timer will slowly fall behind due to ping variance from the server. Sometimes this is worse than others.

It would be helpful to have a speed up button to allow user to sync back up.

**Comment by SilensAngelusNex** [comment]
Seems like this would be due to using the executor's [`scheduleAtFixedRate`](https://github.com/Loze-Put/hunllef-helper/blob/master/src/main/java/com/hunllefhelper/HunllefHelperPlugin.java#L145) to keep track of the time instead of keeping track of server ticks.

Is there a reason the plugin isn't using server ticks currently? Is it so the "2, 1," part of the countdown can occur on the second?

**Comment by Loze-Put** [comment]
@SilensAngelusNex Using ticks would indeed make it a little bit more difficult to time the audio to seconds instead of ticks. However that can be fixed by counting from 0 to 600ms each server tick. The main reason I've never implemented the counter based on ticks is because that would be using game information for the plugin. I think this plugin is already balancing on the edge of what Jagex allows for external PvM plugins. Changing it so it counts based on ingame ticks might make this plugin 'illegal', because it would no longer be possible to recreate the plugin behaviour outside the client. 

Basically I'm very hesitant to add anything that would not be possible by having a website open next to your game client. A button so players can manually alter the timer would be acceptable because it would be possible on an external website.

**Comment by SilensAngelusNex** [comment]
Ah, got it. I'd be a little surprised if counting the server ticks _wasn't_ ok though; you can do similar things with the current metronome.

Is there somewhere I could ask to get an authoritative yes or no?

**Comment by Loze-Put** [comment]
The metronome doesn't do anything PvM related. It doesn't tell you what prayers to use. I think the RuneLite team (through Discord) or a Jmod would be able to make a clear judgement.

**Comment by Loze-Put** [comment]
Closing this as inactive and duplicate of #20.
